### PR TITLE
Implement more subcommands

### PR DESCRIPTION
**Pull request by jwhonce**
* Refactor create subparser to share arguments with run subparser
* Add argparse.*Action subclasses to reduce duplicate code in parsers
* Using BooleanAction now accept True/False value as expected
* .pylintrc added to loosen variable name policing
* Update AbstractBaseAction to remove unset arguments before
  transmitting to podman service
* Align logging messages to podman output
* Renamed global argument from --user to --username, to avoid conflict
  with create/run podman commands
* Add new subcommands: run, create, history, import, info, push,
  restart and search
* Add makefile support for Uploads to PyPi.org
* Add retries to container._refresh()

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+        )
+        parser.add_argument(
+            '--entrypoint',
+            help=('Overwrite the default ENTRYPOINT of the image.'),
+        )
+        parser.add_argument(
+            '--env',
+            '-e',
+            action='append',
+            help=('Set environment variables.'),
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Can be called multiple times.

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+        parser.add_argument(
+            '--env-file',
+            help=('Read in a line delimited file of environment variables,'
+                  ' on the remote host.'),
+        )
+        parser.add_argument(
+            '--expose',
+            metavar='RANGE',
+            help=('Expose a port, or a range of ports'
+                  ' (e.g. --expose=3300-3310) to set up port redirection.'),
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Multiple times.

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+        )
+        parser.add_argument(
+            '--gidmap',
+            metavar='MAP',
+            help=('GID map for the user namespace'),
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: multiple times

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+        )
+        parser.add_argument(
+            '--group-add',
+            action='append',
+            metavar='GROUP',
+            help=('Add additional groups to run as'))
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: multiple times

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+        parser.add_argument('--hostname', help='Container host name')
+
+        volume_group = parser.add_mutually_exclusive_group()
+        volume_group.add_argument(
+            '--image-volume',
+            choices=['bind', 'tmpfs', 'ignore'],
+            metavar='MODE',
+            help='Tells podman how to handle the builtin image volumes')
+        volume_group.add_argument(
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: duplicated call

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+            default=False,
+            help='Keep STDIN open even if not attached. (Default: False)')
+        parser.add_argument('--ipc', help='Create namespace')
+        parser.add_argument(
+            '--kernel-memory', action=UnitAction, help='Kernel memory limit')
+        parser.add_argument(
+            '--label',
+            '-l',
+            help=('Add metadata to a container'
+                  ' (e.g., --label com.example.key=value)'))
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Multiple times

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+        parser.add_argument(
+            '--label-file', help='Read in a line delimited file of labels')
+        parser.add_argument(
+            '--log-driver',
+            choices=['json-file', 'journald'],
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: We don't support journald,

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 8]
```diff
@@ -0,0 +1,394 @@
...
+            '-P',
+            action=BooleanAction,
+            help='Publish all exposed ports to random'
+            ' ports on the host interfaces'
+            '(Default: False)')
+        parser.add_argument(
+            '--quiet',
+            '-q',
+            action='store_true',
+            help='Suppress output information when pulling images')
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Does varlink support sending this information?

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 9]
```diff
@@ -0,0 +1,394 @@
...
+            metavar='MAP',
+            help='Name for GID map from the /etc/subgid file')
+        parser.add_argument(
+            '--subuidname',
+            metavar='MAP',
+            help='Name for UID map from the /etc/subuid file')
+        parser.add_argument(
+            '--sysctl',
+            action='append',
+            help='Configure namespaced kernel parameters at runtime')
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: multiple times

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 10]
```diff
@@ -0,0 +1,394 @@
...
+        parser.add_argument(
+            '--tmpfs', metavar='MOUNT', help='Create a tmpfs mount')
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: multiple times

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 11]
```diff
@@ -0,0 +1,394 @@
...
+        parser.add_argument(
+            '--tty',
+            '-t',
+            action=BooleanAction,
+            default=False,
+            help='Allocate a pseudo-TTY for standard input of container.')
+        parser.add_argument(
+            '--uidmap', metavar='MAP', help='UID map for the user namespace')
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: multiple times

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 12]
```diff
@@ -0,0 +1,394 @@
...
+        parser.add_argument('--ulimit', metavar='OPT', help='Ulimit options')
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: multiple times

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 13]
```diff
@@ -0,0 +1,394 @@
...
+        parser.add_argument(
+            '--user',
+            '-u',
+            help=('Sets the username or UID used and optionally'
+                  ' the groupname or GID for the specified command.'))
+        parser.add_argument(
+            '--userns',
+            choices=['host', 'ns'],
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Do we support container?  
--userns container:CONTAINERID?

I think choices might be incorrect here
--userns ns:PATHTOUSERNS?

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 14]
```diff
@@ -0,0 +1,394 @@
...
+        parser.add_argument(
+            '--oom-kill-disable',
+            action=BooleanAction,
+            help='Whether to disable OOM Killer for the container or not')
+        parser.add_argument(
+            '--oom-score-adj',
+            choices=range(-1000, 1000),
+            metavar='[-1000-1000]',
+            help="Tune the host's OOM preferences for containers")
+        parser.add_argument('--pid', help='Set the PID mode for the container')
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Set the PID Namespace mode fo the container

host, container:UUID, ns:PATH

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 15]
```diff
@@ -0,0 +1,394 @@
...
+            help=('A limit value equal to memory plus swap.'
+                  'Must be used with the --memory flag'))
+        parser.add_argument(
+            '--memory-swappiness',
+            choices=range(0, 100),
+            metavar='[0-100]',
+            help="Tune a container's memory swappiness behavior")
+        parser.add_argument('--name', help='Assign a name to the container')
+        parser.add_argument(
+            '--network',
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: --net is valid also.

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 16]
```diff
@@ -0,0 +1,394 @@
...
+            metavar='BRIDGE',
+            help=('Set the Network mode for the container.'))
```
Comment (rhatdan) [inline contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Set the Network mode for the container

"bridge", host, container:UUID, ns:PATH, none

**Comment by TomSweeneyRedHat** [comment]
I've looked and didn't spot anything over @rhatdan's comments

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py [hunk 17]
```diff
@@ -0,0 +1,394 @@
...
+            'The option can be set multiple times.'
+            '(Format: key=value)')
+        parser.add_argument(
+            '--attach',
+            '-a',
+            action='append',
+            metavar='FD',
+            help=('Attach to STDIN, STDOUT or STDERR. The option can be set'
+                  ' for each of stdin, stdout, and stderr.'))
+
```
[the remark this answers is not in the document]
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Done

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 2 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Fixed.

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 3 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Fixed.

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 4 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: action='append' already in place

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 5 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Updated code so more explicit about the add_mutually_exclusive_group() happening here.

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 6 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Fixed

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 15 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Added.

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 16 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Added

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 8 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Currently always "quiet".  varlink allows progress updates, but that implies the go code is modified to capture the output and proxy it to the client.

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 9 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: action='append'

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 10 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: fixed

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 11 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Fixed

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 12 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: Fixed

File: contrib/python/pypodman/pypodman/lib/actions/_create_args.py (hunk 13 above)
Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/actions/_create_args.py]: I changed this to pass value to service for validation.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**